libraries/botbuilder-azure/src/blobStorage.ts

```diff
@@ -117,14 +117,27 @@ export class BlobStorage implements Storage {
         return this.ensureContainerExists().then((container) => {
             return new Promise<StoreItems>((resolve, reject) => {
                 Promise.all<DocumentStoreItem>(sanitizedKeys.map((key) => {
-                    return new Promise((resolve, reject) => {
-                        this.client.getBlobMetadataAsync(container.name, key).then((blobMetadata) => {
-                            this.client.getBlobToTextAsync(blobMetadata.container, blobMetadata.name).then((result) => {
-                                let document: DocumentStoreItem = JSON.parse(result as any);
-                                document.document.eTag = blobMetadata.etag;
-                                resolve(document);
-                            }, err => resolve(null));
-                        }, (err) => resolve(null));
+                    return this.checkForBlob(container.name, key).then((blobResult) => {
```

Comment:
no need to define a one line function here - just put the check inline (all the other code uses client)

Reply:
Done!